internal/cache/gache/gache_test.go

```diff
@@ -43,41 +52,61 @@ func TestNew(t *testing.T) {
 		afterFunc  func(args)
 	}
 	defaultCheckFunc := func(w want, gotC *cache) error {
-		if !reflect.DeepEqual(gotC, w.wantC) {
+		opts := []cmp.Option{
+			cmp.AllowUnexported(*w.wantC),
+			cmp.AllowUnexported(*gotC),
+			cmp.Comparer(func(want, got *cache) bool {
+				return want.gache != nil && got.gache != nil
```

Comment:
```
opts := []cmp.Option{
	cmp.AllowUnexported(*w.wantC),
	cmp.AllowUnexported(*gotC),
	cmp.Comparer(func(want, got gache.Gache) bool {
		return want != nil && got != nil
	}),
}
if diff := cmp.Diff(w.wantC, gotC, opts...); diff != "" {
	return errors.Errorf("got = %v, want %v", gotC, w.wantC)
}
return nil
```
before: cmp.Comparer(func(want, got *cache) bool {
after: cmp.Comparer(func(want, got gache.Gache) bool {
↑ You can check only gache field with  custom compare function.